var/spack/repos/builtin/packages/py-rpy2/package.py

```diff
@@ -35,13 +35,15 @@ class PyRpy2(PythonPackage):
     homepage = "https://pypi.python.org/pypi/rpy2"
     url = "https://pypi.io/packages/source/r/rpy2/rpy2-2.5.4.tar.gz"
 
-    version('2.5.4', '115a20ac30883f096da2bdfcab55196d')
+    version('2.9.4', '7df2562cdf43a0ccdd1e44ee1c16614f')
     version('2.5.6', 'a36e758b633ce6aec6a5f450bfee980f')
+    version('2.5.4', '115a20ac30883f096da2bdfcab55196d')
 
     # FIXME: Missing dependencies:
     # ld: cannot find -licuuc
     # ld: cannot find -licui18
 
+    depends_on('py-jinja2', type=('build', 'run'))
     depends_on('py-six', type=('build', 'run'))
     depends_on('py-setuptools', type='build')
 
```

Comment:
It looks like Python 2 is no longer supported. Also, many of these dependencies did not exist in earlier versions. You should add some more specific dependency versions:
```python
# All versions
depends_on('py-setuptools', type='build')
depends_on('r',             type=('build', 'run'))

# 2.9.4+
depends_on('r@3.3:',    when='@2.9.4:', type=('build', 'run'))
depends_on('python@3:', when='@2.9.4:', type=('build', 'run'))
depends_on('py-jinja2', when='@2.9.4:', type=('build', 'run'))
depends_on('py-six',    when='@2.9.4:', type=('build', 'run'))

# 2.5.4, 2.5.6
depends_on('r@2.8:',            when='@2.5.4,2.5.6', type=('build', 'run'))
depends_on('py-singledispatch', when='^python@:2',   type=('build', 'run'))
```
That last line might break things, you'll have to test it. It would be nice to know exactly when Python 2 support was dropped so we could add the newest version that does support it.

Reply:
I will add these. I also found https://bitbucket.org/rpy2/rpy2/issues/450/installing-ryp2-with-python-27 which states that 2.8.6 is the latest rpy2 that supports Python2. I will do a bunch of testing and then update the PR.